File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length) { throw new ArgumentException(); }
        if (firstStrand == secondStrand) return 0;

        int distance = 0;
        
        for(int i=0; i< firstStrand.Length; i++)
        {
            if(firstStrand[i] != secondStrand[i]){ ++distance; }
        }

        return distance;
    }
}

File: fsharp/hamming/Hamming.fs
module Hamming

let accumulateDifferencs strandCombination =
    strandCombination |> Seq.sumBy (fun (a,b) -> if a <> b then 1 else 0)

let distance (strand1: string) (strand2: string): int option =
   if strand1.Length = strand2.Length then
     Some (Seq.zip strand1 strand2 |> accumulateDifferencs)
   else
     None                                                   